libraries/standard/http/src/http_client.c

```diff
@@ -929,16 +953,230 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
+        .fieldLen   = fieldLen,
+        .pValueLoc  = pValueLoc,
+        .pValueLen  = pValueLen,
+        .fieldFound = 0u,
+        .valueFound = 0u
+    };
+    size_t numOfBytesParsed = 0u;
+
+    http_parser_init( &parser, HTTP_RESPONSE );
+
+    /* Set the context for the parser. */
+    parser.data = &context;
+
+    /* The intention here to define callbacks just for searching the headers. We will
+     * need to create a private context in httpParser->data that has the field and
+     * value to update and pass back. */
+    http_parser_settings_init( &parserSettings );
+    parserSettings.on_header_field = findHeaderHeaderParsedCallback;
+    parserSettings.on_header_value = findHeaderValueParsedCallback;
+    parserSettings.on_headers_complete = findHeaderOnHeaderCompleteCallback;
+
+    /* Start parsing for the header! */
+    numOfBytesParsed = http_parser_execute( &parser,
+                                            &parserSettings,
+                                            ( char * ) pBuffer,
+                                            bufferLen );
+
+    IotLogDebugWithArgs( "Parsed response for header search: NumBytesParsed=%d",
+                         numOfBytesParsed );
+
+    if( context.fieldFound == 0u )
+    {
+        /* If header field is not found, then both the flags should be zero. */
+        assert( context.valueFound == 0u );
+
+        /* Header is not present in buffer. */
+        IotLogWarnWithArgs( "Header not found in response buffer: "
+                            "RequestedHeader=%.*s", fieldLen, pField );
+
+        returnStatus = HTTP_HEADER_NOT_FOUND;
+    }
+    else if( context.valueFound == 0u )
+    {
+        /* The response buffer is invalid as only the header field was found
+         * in the "<field>: <value>\r\n" format of an HTTP header. */
+        /* IotLogErrorWithArgs( "Unable to find header value in response: " */
+        /*                      "Unexpected state of http-parser: RequestedHeader=%.*s", */
+        /*                      "ParserError=%s", */
+        /*                      fieldLen, pField, */
+        /*                      http_errno_description( HTTP_PARSER_ERRNO( &( parser ) ) ) ); */
+        returnStatus = HTTP_INTERNAL_ERROR;
+    }
+    else
+    {
+        /* Header is found. */
+        assert( ( context.fieldFound == 1u ) && ( context.valueFound == 1u ) );
+
+        IotLogDebugWithArgs( "Found requested header in response: "
+                             "HeaderName=%.*s, HeaderValue=%.*s",
+                             fieldLen, pField,
+                             *pValueLen, *pValueLoc );
+    }
+
+    if( ( returnStatus == HTTP_SUCCESS ) && ( ( parser.http_errno != HPE_CB_header_value ) ) )
+    {
+        IotLogErrorWithArgs( "Header found in response but http-parser returned error: ParserError=%s",
+                             http_errno_description( HTTP_PARSER_ERRNO( &( parser ) ) ) );
+        returnStatus = HTTP_INTERNAL_ERROR;
```

Comment:
Could you please document this error in the HTTPStatus_t enum since you are using it?

Reply:
Added